File: sea-shell-data/src/main/java/org/jesperancinha/shell/webflux/data/SeaShellPersonDto.java
package org.jesperancinha.shell.webflux.data;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Builder
@Setter
@EqualsAndHashCode
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class SeaShellPersonDto {
    @Getter
    private String name;
    @Getter
    private String activity;
    @Getter
    private SeaShellCostumeDto costumeDto;
    @Getter
    private SeaShellAccountDto accountDto;
    private transient Long costumeId;
    private transient String accountId;

    public Long costumeId() {
        return this.costumeId;
    }

    public String accountId() {
        return this.accountId;
    }
}

File: sea-shell-service-spring-web-flux/src/main/kotlin/org/jesperancinha/shell/webflux/data/SeaShellPersonDto.kt
package org.jesperancinha.shell.webflux.data

import lombok.*

@Builder
@Getter
@Setter
@EqualsAndHashCode
@ToString
@AllArgsConstructor
@NoArgsConstructor
class SeaShellPersonDto {
    private val name: String? = null
    private val activity: String? = null
    private val costumeDto: SeaShellCostumeDto? = null
    private val accountDto: SeaShellAccountDto? = null

    @Transient
    private val costumeId: Long? = null

    @Transient
    private val accountId: String? = null
}